RecoTracker/LSTCore/interface/LSTGeometry/ModuleMapMethods.h

```diff
@@ -0,0 +1,294 @@
+#ifndef RecoTracker_LSTCore_interface_LSTGeometry_ModuleMapMethods_h
+#define RecoTracker_LSTCore_interface_LSTGeometry_ModuleMapMethods_h
+
+#include <cmath>
+#include <cassert>
+#include <iterator>
+#include <vector>
+#include <tuple>
+#include <unordered_set>
+#include <unordered_map>
+#include <initializer_list>
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/polygon.hpp>
+
+#include "LSTMath.h"
+#include "RecoTracker/LSTCore/interface/LSTGeometry/LSTMath.h"
+#include "RecoTracker/LSTCore/interface/LSTGeometry/Centroid.h"
+#include "RecoTracker/LSTCore/interface/LSTGeometry/Module.h"
+#include "RecoTracker/LSTCore/interface/LSTGeometry/DetectorGeometry.h"
+
+namespace lstgeometry {
+
+  std::vector<unsigned int> getStraightLineConnections(unsigned int ref_detid,
+                                                       std::unordered_map<unsigned int, Centroid> const& centroids,
+                                                       DetectorGeometry const& det_geom) {
+    auto centroid = centroids.at(ref_detid);
+
+    double refphi = std::atan2(centroid.y, centroid.x);
+
+    Module refmodule(ref_detid);
+
+    unsigned short ref_layer = refmodule.layer();
+    unsigned short ref_subdet = refmodule.subdet();
+
+    auto const& tar_detids_to_be_considered =
+        ref_subdet == 5 ? det_geom.getBarrelLayerDetIds(ref_layer + 1) : det_geom.getEndcapLayerDetIds(ref_layer + 1);
+
+    std::vector<unsigned int> list_of_detids_etaphi_layer_tar;
+    for (unsigned int tar_detid : tar_detids_to_be_considered) {
+      if (moduleOverlapsInEtaPhi(det_geom.getCorners(ref_detid), det_geom.getCorners(tar_detid), refphi, 0) ||
+          moduleOverlapsInEtaPhi(det_geom.getCorners(ref_detid), det_geom.getCorners(tar_detid), refphi, 10) ||
+          moduleOverlapsInEtaPhi(det_geom.getCorners(ref_detid), det_geom.getCorners(tar_detid), refphi, -10))
+        list_of_detids_etaphi_layer_tar.push_back(tar_detid);
+    }
+
+    // Consider barrel to endcap connections if the intersection area is > 0
+    if (ref_subdet == 5) {
+      std::unordered_set<unsigned int> barrel_endcap_connected_tar_detids;
+
+      for (int zshift : {0, 10, -10}) {
+        std::vector<Polygon> ref_polygon;
+        ref_polygon.push_back(getEtaPhiPolygon(det_geom.getCorners(ref_detid), refphi, zshift));
+
+        // Check whether there is still significant non-zero area
```

Comment:
it would help to add some more detailed description of what is done (it took me a while to stare at this code to figure out what's happening)